opentelemetry-api/src/opentelemetry/attributes/__init__.py

```diff
@@ -173,30 +267,35 @@ def __init__(
     def __repr__(self) -> str:
         return f"{dict(self._dict)}"
 
-    def __getitem__(self, key: str) -> types.AttributeValue:
+    def __getitem__(self, key: str) -> types.AnyValue:
         return self._dict[key]
 
-    def __setitem__(self, key: str, value: types.AttributeValue) -> None:
+    def __setitem__(self, key: str, value: types.AnyValue) -> None:
         if getattr(self, "_immutable", False):  # type: ignore
             raise TypeError
         with self._lock:
             if self.maxlen is not None and self.maxlen == 0:
                 self.dropped += 1
                 return
 
-            value = _clean_attribute(key, value, self.max_value_len)  # type: ignore
-            if value is not None:
-                if key in self._dict:
-                    del self._dict[key]
-                elif (
-                    self.maxlen is not None and len(self._dict) == self.maxlen
-                ):
-                    if not isinstance(self._dict, OrderedDict):
-                        self._dict = OrderedDict(self._dict)
-                    self._dict.popitem(last=False)  # type: ignore
-                    self.dropped += 1
-
-                self._dict[key] = value  # type: ignore
+            if self._extended_attributes:
+                value = _clean_extended_attribute(
+                    key, value, self.max_value_len
+                )
+            else:
+                value = _clean_attribute(key, value, self.max_value_len)  # type: ignore
+                if value is None:
+                    return
```

Comment:
Can the value be `None` in the other branch on L282?

Reply:
In the other branch None is fine because it's a valid AnyValue